### PR TITLE
Fix (core, backend): Revert operation - consider location of root git folder

### LINKED ISSUE #427: 'Revert Changes' isn't working on a standalone site deployed on Platfrom.sh
## Description
**Steps to reproduce:**
 1. deploy the test site
 2. open Edit env, switch to Edit Mode
 3. make some changes in the content
 4. click 'Revert'
 *Actual Results:*
 'An error has occurred' message shows

**Expected Results:**
 Operation executes with no errors.

==========
 **Original Issue:**

```*Steps to reproduce:*
 1. execute:
 git clone https://github.com/johnsonandjohnson/Bodiless-JS
 2. execute:
 cd Bodiless-JS
 3. execute:
 npm run setup
 4. execute:
 npm run sites:launch-test
 5. open any page, e.g. /touts/
 6. add some content
 7. click Revert Changes icon, confirm reverting
 8. Click Ok on a browser alert
 9. Refresh the page
 *Actual Results:*
 Content is not removed
 *Expected Results:*
 Content should be cleaned
 *+NOTE:+*
 Revert changes' works as expected if a test site is launched from examples/test-site/ via 'npm run start'

```

### PR DESCRIPTION
**Pull request by konst8**
## Changes
Make git revert operation considering location of .git folder and resolve path base on that.
This is important when launching the site outside the monorepo.

## Testing notes
- launch starter site by running `npm run sites:launch-test` or in a standalone folder
- init clean git repo in the site folder, add and commit everything
- try adding and reverting new pages in edit mode

## Related Issues
Fixes #427

### CODE REVIEW

File: packages/bodiless-backend/src/backend.js [hunk 1]
```diff
@@ -362,13 +362,21 @@ class Backend {
...
+            .exec();
           const reGetDeletedAndUntracked = /(?<= D |\?\? ).*/gm;
           const deletedAndUntracked = gitStatus.stdout.match(reGetDeletedAndUntracked);
           if (deletedAndUntracked !== null) {
             const dataPagePath = path.join(backendFilePath, 'pages');
             const obsoletePublicPages = deletedAndUntracked.map(gitPath => {
               const publicPagePath = gitPath.replace(dataPagePath, backendPublicPath);
-              return path.resolve('../..', publicPagePath);
+              // Get absolute path considering location of .git folder
+              const absolutePublicPagePath = path.resolve(
```
Comment (dewen) [inline packages/bodiless-backend/src/backend.js:375]: can we just return from here?

**Comment by TatsianaMiraniuk** [comment]
@konst8 
that seems to be working unstable. After deploying the env, updating some content on Home page and then doing the revert the very first time, I got an error:
Screen Shot 2020-07-10 at 1 46 05 PM

After next attempts, the revert changes seemed to be executed and showed 'revert completed', but after reloading the page it looked incorrectly, e.g.
1) In Main Menu, there 5 items are populated. One test item was added. After reverting the changes and reloading the page, only 1 item was left (first item - Home). The remaining 4 default items and 1 that was added as a new, they all disappeared.
However after some time, 4 default items got back.  That was caught at least once.
2) after trying to reproduce the issue above, the following was found. In Main menu, when adding 1 item, and then doing the revert changes, that 1 item was removed as expected in the page, but after opening another page, that item wasn't removed.
Screen Shot 2020-07-10 at 2 01 47 PM

Can you please take a look?

Thanks,
Tatsiana

**Comment by konst8** [comment]
@TatsianaMiraniuk, regarding the error "The application encountered an issue ...". This message is shown when 10 secs timeout is over and can possibly happen on the "create page" operation. Although, I was not able to reproduce it.
The same with the issue 1 - wasn't able to recreate neither locally, nor on psh.
Issue 2 happened to me a couple of times on psh. After reverting chagnes and opening another page, the new menu items were still shown, but they disappeared after a second or two. On another attempt, the items disappeared when I hovered on them. This looks like some kind of cache or watch-delay issue. And I think it deserves a new ticket.

**Comment by hvanyo** [comment]
> @TatsianaMiraniuk, regarding the error "The application encountered an issue ...". This message is shown when 10 secs timeout is over and can possibly happen on the "create page" operation. Although, I was not able to reproduce it.
> The same with the issue 1 - wasn't able to recreate neither locally, nor on psh.
> Issue 2 happened to me a couple of times on psh. After reverting chagnes and opening another page, the new menu items were still shown, but they disappeared after a second or two. On another attempt, the items disappeared when I hovered on them. This looks like some kind of cache or watch-delay issue. And I think it deserves a new ticket.

Opened -> https://github.com/johnsonandjohnson/Bodiless-JS/issues/442 for second issue

**Pull request merged**